test/fluree/http_api/system_test.clj

```diff
@@ -161,8 +168,183 @@
                                          :where  '[[?t :type :schema/Test]]}})
                        :headers edn-headers}
           query-res   (post :query query-req)]
-      (is (= 200 (:status query-res)))
+      (is (= 200 (:status query-res))
+          (str "Response was: " (pr-str query-res)))
       (is (= [{:id       :ex/query-test
                :rdf/type [:schema/Test]
                :ex/name  "query-test"}]
-             (-> query-res :body edn/read-string))))))
+             (-> query-res :body edn/read-string)))))
+
+  (testing "can query a basic entity w/ JSON"
+    (let [ledger-name  (create-rand-ledger "query-endpoint-basic-entity-test")
+          json-headers {"Content-Type" "application/json"
+                        "Accept"       "application/json"}
+          txn-req      {:body
+                        (json/write-value-as-string
+                          {:ledger ledger-name
+                           :txn    [{:id      :ex/query-test
+                                     :type    :schema/Test
+                                     :ex/name "query-test"}]})
+                        :headers json-headers}
+          txn-res      (post :transact txn-req)
+          _            (assert (= 200 (:status txn-res)))
+          query-req    {:body
+                        (json/write-value-as-string
+                          {:ledger ledger-name
+                           :query  {:select '{?t [:*]}
+                                    :where  '[[?t :type "schema/Test"]]}})
+                        :headers json-headers}
+          query-res    (post :query query-req)]
+      (is (= 200 (:status query-res))
+          (str "Response was: " (pr-str query-res)))
+      (is (= [{:id       "ex/query-test"
+               :rdf/type ["schema/Test"]
+               :ex/name  "query-test"}]
+             (-> query-res :body
+                 (json/read-value json/keyword-keys-object-mapper)))))))
```

Comment:
It might be useful to have just plain old `json/read-value` here instead of trying to coerce it back into keywords. I'm not sure this `:id` value is correct - if it's json it should be `ex:query-test`, I think, or if we're converting it to keywords then the `:id` should be `:ex/query-test`.